### ISSUE TITLE
"registry.terraform.io/hashicorp/azurerm" produced an invalid new value

### ISSUE THREAD

**Issue by cimnine** [hashicorp/terraform-provider-azurerm#14873]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request

### Terraform (and AzureRM Provider) Version

* Terraform Core version: 1.1.3
* AzureRM Provider version: v2.91.0

### Terraform Configuration Files

```hcl
resource "azurerm_linux_web_app" "xxx" {
  name                = format("xxx-%s", var.customer_name)
  resource_group_name = data.azurerm_resource_group.xxx.name
  location            = data.azurerm_resource_group.xxx.location
  service_plan_id     = data.azurerm_service_plan.app_service_plan.id

  https_only = true

  identity {
    type = "SystemAssigned"
  }

  auth_settings {
    enabled = false
  }

  site_config {
    always_on          = local.is_production
    http2_enabled      = true
    ftps_state         = "Disabled"
    websockets_enabled = true

    use_32_bit_worker = false
    worker_count      = local.is_production ? var.xxx_worker_count : 1

    container_registry_use_managed_identity = false

    # TODO health_check_path = ""

    application_stack {
      docker_image     = var.xxx_docker_image
      docker_image_tag = var.xxx_docker_image_tag
    }
  }
}
```

[The full project as GPG-encrypted ZIP.](https://github.com/hashicorp/terraform-provider-azurerm/files/7837606/code.gpg.zip) (You may need to change the file ending back to `code.zip.gpg`, as GitHub would not let me attach a `.gpg` file.)

### Description / Feedback

I've successfully applied the above resource, but Terraform reported the following error:

```
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.xxx_instance["dev"].module.xxx_customer["xxx"].azurerm_linux_web_app_slot.tacs3_preview
│ to include new values learned so far during apply, provider
│ "registry.terraform.io/hashicorp/azurerm" produced an invalid new value for
│ .site_config[0].use_32_bit_worker: was cty.False, but now cty.True.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
```

After applying once more the inconsistency was rectified.

**Comment by ms-henglu** [comment]
It seems that it can't set `use_32_bit_worker = false` when create a resource, but it can be done when update it.

Let me check with service team whether this is by design.

**Comment by ms-henglu** [comment]
It seems that it cannot set `use_32_bit_worker=false` when create this resource, but can update it.

I've raised a PR to fix it.

**Comment by jackofallops** [comment]
Hi @cimnine - I believe this was addressed in #14940 so I'm going to close this issue for now.  If you believe it to still be a problem, we can look to re-open.

Thanks!
